api/src/services/service_utils.py

```diff
@@ -77,3 +77,29 @@ def _add_search_filters(
                 start_date,
                 end_date,
             )
+
+
+def _add_top_level_agency_prefix(
```

Comment:
This function doesn't feel like it should be in this generic util as it's very specific to the use case of the search opportunities. It expects filters to basically be a specific type, but you're having it take in a generic type and then doing several hasattr checks. I'd just move this function to the search opportunities logic and not worry about making it generic.